### DISCUSSION TITLE
Error when exporting vtk files

### DISCUSSION THREAD

**Discussion by Devin-one** [PMEAL/porespy#820] [Q&A]
Hello, Professor. I was getting an error when exporting a vtk file. 
![image](https://user-images.githubusercontent.com/93040662/230712294-8585a688-4c34-407a-b905-79ea523da4f5.png)

Here is my code. 
```
import numpy as np
import porespy as ps
import openpnm as op
import matplotlib.pyplot as plt
import inspect
inspect.signature(ps.metrics.two_point_correlation)
from skimage import io
ps.visualization.set_mpl_style()
import imageio
import scipy as sp

im1 = imageio.imread('D:\Spyderfile\test.tif')
im=255-im1
im = im >= 255
snow_output = ps.networks.snow2(im, voxel_size=1)
pn = op.io.network_from_porespy(snow_output.network)
pn.update(snow_output)
prj = pn.project
prj.export_data(filename='extraction_1', filetype='vtk')
im = ps.tools.align_image_with_openpnm(im).astype(int)
ps.io.to_vtk(sp.array(im, dtype=int)[:, :, np.newaxis], 'im')
```
However, when I added one of the lines of code, `snow_output = ps.networks.snow2(im, voxel_size=1,boundary_faces=['left','right','front','back'],marching_cubes_area=False)`, another`` error appeared.
![image](https://user-images.githubusercontent.com/93040662/230712212-0474ad7b-fe33-42eb-9773-60dd0ca9f915.png)
I don't know what went wrong. Thank you in advance for your reply.

**Comment by jgostick** [comment]
you are using code for old versions of porespy and openpnm, but using newer versions.  You can refer to our docs websites for examples on how to use the new versions.

**Comment by Devin-one** [comment]
I'm sorry I can't view the online examples provided in the IO module.
![1681179005855](https://user-images.githubusercontent.com/93040662/231037987-bd9592c1-64d4-433f-9748-9b2f83efd883.jpg)

**Reply by jgostick** [reply]
The first error is coming when you try to 'update' the openpnm network dict, so I was refering to the fact that you should be looking at the new OpenPNM docs.

**Comment by Devin-one** [comment]
Thank you for your kind reminder. I successfully exported the file. But when I rendered images in TIF format and networks in Paraview, they did not correspond correctly. It flipped the original image vertically when the viewer was afraid to open the TIF image. 
Here is my TIF image.
![p33](https://user-images.githubusercontent.com/93040662/231134320-76bf98a6-75a1-42b9-8bd1-0e67a473d240.png)
The following is the image rendered by  paraview.
![image](https://user-images.githubusercontent.com/93040662/231137049-db7a7763-7184-406a-881b-3824a4d2d681.png)

Here is my code. Please point out the key points.
```
snow_output = ps.networks.snow2(
    im, voxel_size=1
)
im = ps.tools.align_image_with_openpnm(im).astype(int)
snow = ps.filters.snow_partitioning(im)
regions = snow.regions
imageio.imsave("p4.tif", np.array(regions.astype("uint8")))
path_to_file = "D:\Spyderfile\p0"
op.io._vtk.project_to_vtk(pn.project, filename=path_to_file+'/extraction')
```

**Reply by jgostick** [reply]
use porespy.io.to_vtk to export your image, and it will match openpnm inside paraview.
